bokehjs/src/lib/models/widgets/tables/cell_formatters.ts

```diff
@@ -71,6 +71,11 @@ export class StringFormatter extends CellFormatter {
   override doFormat(_row: any, _cell: any, value: any, _columnDef: any, dataContext: any): string {
     const {font_style, text_align, text_color, background_color} = this
 
+    // unfortunately isNaN("foo") is true
+    if (value == null || (isNumber(value) && isNaN(value))) {
```

Comment:
I mentioned this in a recent PR that `isNaN()` must not be used without a type guard, because `isNaN("foo")` equivalent to `isNaN(Numer("foo"))` and `Number("foo")` is literal `NaN`. There's `Number.isNaN()` that behaves better, so you may want to use that to simplify the code and make the comment unnecessary. For completeness, one can also use `Object.is()` to compare with `NaN`, e.g.:
```ts
> Object.is(Number("foo"), NaN)
true
```
though I wouldn't prefer this over `Number.isNaN()`.